addon/components/address-plugin/insert.ts

```diff
@@ -0,0 +1,103 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { SayController } from '@lblod/ember-rdfa-editor';
+import { tracked } from 'tracked-built-ins';
+import { restartableTask, task, timeout } from 'ember-concurrency';
+
+import { AddressSuggestion } from './types';
+import { getAddressMatch, getSuggestedLocations } from './utils';
+
+type Args = {
+  controller: SayController;
+  options: {
+    endpoint: string;
+  };
+};
+
+export default class AddressPluginInsertComponent extends Component<Args> {
+  @tracked modalOpen = false;
+  @tracked isSelectingAddress = false;
+
+  @tracked addressSuggestions: AddressSuggestion[] = [];
+  @tracked selectedAddress: AddressSuggestion | null = null;
+
+  get controller() {
+    return this.args.controller;
+  }
+
+  get schema() {
+    return this.controller.schema;
+  }
+
+  @action
+  openModal() {
+    this.modalOpen = true;
+  }
+
+  @action
+  closeModal() {
+    this.modalOpen = false;
+    this.selectedAddress = null;
+  }
+
+  @action
+  selectAddress(selection: AddressSuggestion) {
+    this.selectedAddress = selection;
+  }
+
+  get canInsertAddress() {
+    if (this.isSelectingAddress) {
+      return false;
+    }
+
+    if (!this.selectedAddress) {
+      return false;
+    }
+
+    return true;
+  }
+
+  insertAddress = task(async () => {
+    if (this.selectedAddress) {
+      this.isSelectingAddress = true;
+
+      const addresses = await getAddressMatch({
+        housenumber: this.selectedAddress.Housenumber,
+        municipality: this.selectedAddress.Municipality,
+        street: this.selectedAddress.Thoroughfarename,
+        zipcode: this.selectedAddress.Zipcode,
+      });
+
+      const address = addresses[0];
+
+      this.controller.withTransaction((tr) => {
+        tr.replaceSelectionWith(
+          this.schema.node(
+            'block_rdfa',
+            {
+              typeof: 'https://data.vlaanderen.be/ns/adres#Adres',
+              resource: address.identificator.id,
+            },
+            this.schema.node('paragraph', null, [
```

Comment:
bit awkward to have adresses always be a block
also it's incredibly easy to nest adresses, leading to nonsensical rdfa

I know it's a PoC, and the focus was the endpoint logic so I suppose its good enough

Reply:
no then its still a block
we'll likely need a custom type